fl4health/clients/deep_mmd_clients/ditto_deep_mmd_client.py

```diff
@@ -103,9 +114,95 @@ def update_before_train(self, current_server_round: int) -> None:
         )
         # Register hooks to extract features from the initial global model if not already registered
         self.initial_global_feature_extractor._maybe_register_hooks()
-        # Enable training of Deep MMD loss layers
-        for layer in self.flatten_feature_extraction_layers.keys():
-            self.deep_mmd_losses[layer].training = True
+        # Enable training of Deep MMD loss layers if the beta_global_update_interval is set to -1
+        # meaning that the betas will be updated after each individual batch based on only that
+        # individual batch
+        if self.beta_global_update_interval == -1:
+            for layer in self.flatten_feature_extraction_layers.keys():
+                self.deep_mmd_losses[layer].training = True
+
+    def _should_optimize_betas(self, step: int) -> bool:
+        step_at_interval = (step - 1) % self.beta_global_update_interval == 0
+        valid_components_present = self.initial_global_model is not None
+        # If the Deep MMD loss doesn't matter, we don't bother optimizing betas
+        weighted_deep_mmd_loss = self.deep_mmd_loss_weight != 0
+        return step_at_interval and valid_components_present and weighted_deep_mmd_loss
+
+    def update_after_step(self, step: int, current_round: Optional[int] = None) -> None:
+        if self.beta_global_update_interval > 0 and self._should_optimize_betas(step):
+            # Get the feature distribution of the local and initial global features with evaluation
+            # mode
+            local_distributions, initial_global_distributions = self.update_buffers(
+                self.model, self.initial_global_model
+            )
+            # Update betas for the Deep MMD loss based on gathered features during training
+            for layer, layer_deep_mmd_loss in self.deep_mmd_losses.items():
+                layer_deep_mmd_loss.training = True
+                _ = layer_deep_mmd_loss(local_distributions[layer], initial_global_distributions[layer])
```

Comment:
I don't think you need to "catch" this return value if you're not going to use it. You can just call the loss.